packages/elastic-analytics/src/events/types.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { ShipperName } from '../analytics_client';
+
+export interface EventContext {
+  // TODO: Extend with known keys
+  [key: string]: unknown;
+}
+
+/**
+ * Event Type used for indexed structures. Only used to improve the readability of the types
+ */
+export type EventType = string;
+
+/**
+ * Types of the Telemetry Counter: It allows to differentiate what happened to the events
+ */
+export enum TelemetryCounterType {
+  /**
+   * The event was accepted and will be sent to the shippers when they become available (and opt-in === true).
+   */
+  enqueued = 'enqueued',
+  /**
+   * The event was sent to at least one shipper.
+   */
+  sent_to_shipper = 'sent_to_shipper',
+  /**
+   * The event was successfully sent by the shipper.
+   */
+  succeeded = 'succeeded',
+  /**
+   * There was an error when processing/shipping the event.
+   * Refer to the Telemetry Counter's code for the reason.
+   */
+  failed = 'failed',
+  /**
+   * The event was dropped from the queue.
+   * Refer to the Telemetry Counter's code for the reason.
+   */
+  dropped = 'dropped',
+}
+
+/**
+ * Shape of the events emitted by the telemetryCounter$ observable
+ */
+export interface TelemetryCounter {
+  /**
+   * Indicates if the event contains data about succeeded, failed or dropped events.
+   */
+  type: TelemetryCounterType;
+  /**
+   * Who emitted the event? It can be "client" or the name of the shipper.
+   */
+  source: 'client' | ShipperName;
```

Comment:
NIT: given `client` is a special value here, I wonder if we should be forbidding to register a shipper with the `client` name.